Corona-Warn-App/src/main/java/de/rki/coronawarnapp/contactdiary/ui/edit/ContactDiaryEditLocationsViewModel.kt

```diff
@@ -0,0 +1,51 @@
+package de.rki.coronawarnapp.contactdiary.ui.edit
+
+import androidx.lifecycle.MediatorLiveData
+import androidx.lifecycle.asLiveData
+import com.squareup.inject.assisted.AssistedInject
+import de.rki.coronawarnapp.contactdiary.model.ContactDiaryLocation
+import de.rki.coronawarnapp.contactdiary.storage.repo.ContactDiaryRepository
+import de.rki.coronawarnapp.ui.SingleLiveEvent
+import de.rki.coronawarnapp.util.coroutine.DispatcherProvider
+import de.rki.coronawarnapp.util.viewmodel.CWAViewModel
+import de.rki.coronawarnapp.util.viewmodel.SimpleCWAViewModelFactory
+
+class ContactDiaryEditLocationsViewModel @AssistedInject constructor(
+    private val contactDiaryRepository: ContactDiaryRepository,
+    dispatcherProvider: DispatcherProvider
+) : CWAViewModel(dispatcherProvider = dispatcherProvider){
+
+    val locationsLiveData = contactDiaryRepository.locations.asLiveData()
+
+    val navigationEvent = SingleLiveEvent<NavigationEvent>()
+
+    val isButtonEnabled = MediatorLiveData<Boolean>().apply {
+        addSource(locationsLiveData) {
+            value = !it.isNullOrEmpty()
+        }
+    }
```

Comment:
couldnt you map the flow instead and call asLiveData on the map?

```suggestion
    val isButtonEnabled = locationsLiveData.map { it.isNullOrEmpty() }.asLiveData()
```

Reply:
No that line does not work, it's not a flow

Reply:
then do it before the conversion to live data? use `contactDiaryRepository.locations` directly?

Reply:
private val locationsFlow = contactDiaryRepository.locations
val locationsLiveData = locationsFlow.asLiveData()
val isButtonEnabled = locationsFlow.map { it.isNotEmpty() }.asLiveData() // it can't be null in this case